plugins/woocommerce/tests/legacy/unit-tests/order/class-wc-tests-order-functions.php

```diff
@@ -54,10 +54,23 @@ public function test_wc_is_order_status() {
 	 * Test wc_get_order_status_name().
 	 *
 	 * @since 2.3.0
+	 * @expectedIncorrectUsage wc_get_order_status_name
 	 */
 	public function test_wc_get_order_status_name() {
 		$this->assertEquals( _x( 'Pending payment', 'Order status', 'woocommerce' ), wc_get_order_status_name( OrderInternalStatus::PENDING ) );
 		$this->assertEquals( _x( 'Pending payment', 'Order status', 'woocommerce' ), wc_get_order_status_name( OrderStatus::PENDING ) );
+
+		$this->assertEquals(
+			'Unexpected and unknown',
+			wc_get_order_status_name( 'Unexpected and unknown' ),
+			'Strings that do not map to a known status will be returned unmodified.'
```

Comment:
But I see, we don't allow status which are not strictly mapped, where is this defined?